identity-android/src/main/java/com/android/identity/android/direct_access/DirectAccess.kt

```diff
@@ -0,0 +1,507 @@
+     * existing credential in the slot.
+     *
+     * @param documentSlot the slot to provision a credential in
+     * @return signingKeyCertificate, encryptedPresentationData
+     */
+    fun createCredential(
+        documentSlot: Int
+    ): Pair<X509CertChain, ByteArray> {
+        val response: ByteArray?
+        try {
+            transport.closeConnection()
+            transport.openConnection()
+
+            val notBefore = System.currentTimeMillis()
+            val notAfter = System.currentTimeMillis() + (365 * 24 * 60 * 60 * 1000).toLong() // todo @venkat the validity duration should be set during certification as opposed to creation
+            val scratchpad = ByteArray(2)
+            val bos = ByteArrayOutputStream()
+            // set instruction
+            setShort(scratchpad, 0, CMD_MDOC_CREATE_PRESENTATION_PKG.toShort())
+            bos.write(scratchpad)
+            bos.write(documentSlot)
+            val notBeforeBytes = encodeValidityTime(notBefore)
+            val notAfterBytes = encodeValidityTime(notAfter)
+
+            // Set Not Before
+            setShort(scratchpad, 0, notBeforeBytes.size.toShort())
+            bos.write(scratchpad)
+            bos.write(notBeforeBytes)
+
+            // Set Not After
+            setShort(scratchpad, 0, notAfterBytes.size.toShort())
+            bos.write(scratchpad)
+            bos.write(notAfterBytes)
+            val apdu = makeCommandApdu(bos.toByteArray())
+
+            response = transport.sendData(apdu)
+            val status = getAPDUResponseStatus(response)
+            check(APDU_RESPONSE_STATUS_OK == status) {
+                "createPresentationPackage failed. Response status: $status" }
+            val input = response.copyOf(response.size - 2)
+
+            var signingCert: X509CertChain? = null
+            var encryptedData: ByteArray? = null
+            val map = decode(input)
+            val keys = map.asMap.keys
+            for (keyItem in keys) {
+                val value = keyItem.asNumber.toInt()
+                when (value) {
+                    KEY_CERT -> {
+                        val bStrItem = map[keyItem]
+                        val certData = bStrItem.asBstr
+                        var credentialKeyCert: List<X509Cert?>? = null
+                        try {
+                            val cf = CertificateFactory.getInstance("X.509")
+                            val bis = ByteArrayInputStream(certData)
+                            credentialKeyCert = listOf(X509Cert((cf.generateCertificate(bis) as X509Certificate).encoded))
+                        } catch (e: CertificateException) {
+                            throw IllegalStateException(
+                                "Error generating signing certificate from response",
+                                e
+                            )
+                        }
+                        signingCert = X509CertChain(credentialKeyCert)
```

Comment:
The detour to CertificateFactory seems weird, why can't L297-L310 simply be replaced with
```
signingCert = X509CertChain(listOf(X509Cert(map[keyItem].asBstr))
```